src/Pages/EmailList/EmailList.js

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import './email-list.css';
+import { getAllUsers } from '../../APIFunctions/User';
+import {
+  Button,
+  ButtonDropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from 'reactstrap';
+
+export default class EmailList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      users: [],
+      filteredUsers: [],
+      filtered: false,
+      dropdownOpen: false
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.user) {
+      this.getUsers();
+    }
+  }
+
+  handleToggle = () => {
+    this.setState({ dropdownOpen: !this.state.dropdownOpen });
+  }
+
+  getUsers = async () => {
+    const apiResponse = await getAllUsers(this.props.user.token);
+    if (!apiResponse.error){
+      let userData = apiResponse.responseData;
+      let users = [];
+      for(let i=0; i<userData.length; i++){
+        users.push({
+          email: userData[i].email,
+          accessLevel: userData[i].accessLevel,
+          membershipValidUntil: userData[i].membershipValidUntil,
+          emailOptIn: userData[i].emailOptIn
+        });
+      }
+      this.setState({users});
+    }
+  }
+
+  /*
+  Filters users via filterID parameter
+    0: Non-Valid Members
+    1: Members
+    2: Officers
+    3: Admin
+    4: Everyone Valid
+  */
+  filterUsers = (filterID) => {
```

Comment:
change this function to use the `membershipState` enum from [`src/Enums.js`](https://github.com/SCE-Development/Core-v4/blob/addd3e75b606aeefca6cfef47d7d4d56dc6810b8/src/Enums.js#L34)

Reply:
The membershipState enum from that file doesn't have the label for filtering everyone valid.

Reply:
True. I was thinking we could abstract this so the file doesn't look to complex. Let me know what you think about this:

1. Define an Enum in [``](https://github.com/SCE-Development/Core-v4/blob/dev/src/Enums.js)
```js
const userFilterType = {
  VALID: 0,
  NON_VALID: 1,
  ALL: 2
};
```

1. We move the function to [`src/APIFunctions/User.js`](https://github.com/SCE-Development/Core-v4/blob/addd3e75b606aeefca6cfef47d7d4d56dc6810b8/src/APIFunctions/User.js):
```js
	    let filteredUsers = this.state.users.filter((user) => {
      const d = new Date();
      if(filterID === userFilterType.VALID){ // Everyone valid
        if(user.accessLevel>=MEMBER &&
               user.membershipValidUntil > d.toISOString()){
          return user;
        }
      } else if(filterID === userFilterType.NON_VALID){ // Non-valid/Pending
        if(user.accessLevel === NON_MEMBER || user.accessLevel === PENDING){
          return user;
        }
      } else{ // Member/Officer/Executive
        if(user.accessLevel === filterID  &&
               user.membershipValidUntil > d.toISOString()){
          return user;
        }
      }
      return '';
    });
    return filteredUsers;
  }
```